### PR TITLE
Adds PGSQL Policies for  Single Server and Flexible Server:

### PR DESCRIPTION
**Pull request by sasever**
PostgreSQL  FSI LZ policies for Single Server and Flexible server:

Implemented Customs:

1. Deny-Server-PostgreSql-Ssl
2. Deny-PostgreSql-Server-Tls-Minimum-Version
3. Deny-PostgreSql-Flexible-Server-Customer-Managed-Keys
4. Deny-PostgreSql-Server-Flexible-Server-LogAnalytics

Referred Built-ins:

1. Dine-PostgreSQL-Server-Adv-Threat-Protection
2. Deny-PostgreSql-Server-Public-Network-Access
3. Deny-PostgreSql-Flexible-Public-Network-Access
4. Deny-PostgreSql-Server-Infrastructure-Encryption

## PR Checklist

* [x ] Code of Conduct signed. If not, go to [Code of Conduct](https://github.com/microsoft/industry/blob/main/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json [hunk 1]
```diff
@@ -17,47 +17,343 @@
...
+                                                "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+                                                "contentVersion": "1.0.0.0",
+                                                "parameters": {
+                                                    "resourceName": { "type": "String" },
+                                                    "resourceType": { "type": "String" },
+                                                    "logAnalytics": { "type": "String" },
+                                                    "location": { "type": "String" },
+                                                    "profileName": { "type": "String" },
+                                                    "metricsEnabled": { "type": "String" },
+                                                    "logsEnabled": { "type": "String" }
```
Comment (marvinbuss) [inline foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json:276]: lint

File: foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json [hunk 2]
```diff
@@ -17,47 +17,343 @@
...
+                                                        "name": "[[concat(parameters('resourceName'), '/', 'Microsoft.Insights/', parameters('profileName'))]]",
+                                                        "location": "[[parameters('location')]]",
+                                                        "dependsOn": [],
+                                                        "properties": {
+                                                            "workspaceId": "[[parameters('logAnalytics')]]",
+                                                            "metrics": [
+                                                                {
+                                                                    "category": "AllMetrics",
+                                                                    "enabled": "[[parameters('metricsEnabled')]]",
+                                                                    "retentionPolicy": { "days": 0, "enabled": false },
```
Comment (marvinbuss) [inline foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json:293]: lint

File: foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json [hunk 3]
```diff
@@ -17,47 +17,343 @@
...
+                                                                },
+                                                                {
+                                                                    "category": "QueryStoreRuntimeStatistics",
+                                                                    "enabled": "[[parameters('logsEnabled')]]"
+                                                                },
+                                                                {
+                                                                    "category": "QueryStoreWaitStatistics",
+                                                                    "enabled": "[[parameters('logsEnabled')]]"
+                                                                }
+                                                            ]
```
Comment (marvinbuss) [inline foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json:335]: For logs you can also use categoryGroups these days. Can you please change that?

File: foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json [hunk 4]
```diff
@@ -17,47 +17,343 @@
...
+                                                                    "retentionPolicy": { "days": 0, "enabled": false },
+                                                                    "timeGrain": null
+                                                                }
+                                                            ],
+                                                            "logs": [
+                                                                {
+                                                                    "category": "PostgreSQLLogs",
+                                                                    "enabled": "[[parameters('logsEnabled')]]"
+                                                                }
+                                                            ]
```
Comment (marvinbuss) [inline foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json:302]: For logs you can also use categoryGroups these days. Can you please change that?

File: foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json [hunk 5]
```diff
@@ -17,47 +17,343 @@
...
+                                                "outputs": {}
+                                            },
+                                            "parameters": {
+                                                "logAnalytics": { "value": "[[parameters('logAnalytics')]]" },
+                                                "location": { "value": "[[field('location')]]" },
+                                                "resourceName": { "value": "[[field('name')]]" },
+                                                "resourceType": { "value": "[[field('type')]]" },
+                                                "profileName": { "value": "[[parameters('profileName')]]" },
+                                                "metricsEnabled": { "value": "[[parameters('metricsEnabled')]]" },
+                                                "logsEnabled": { "value": "[[parameters('logsEnabled')]]" }
```
Comment (marvinbuss) [inline foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json:348]: lint

File: foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json [hunk 6]
```diff
@@ -126,54 +422,110 @@
...
+                                        {
+                        "policyDefinitionId": "[concat(variables('scope'), '/providers/Microsoft.Authorization/policyDefinitions/', variables('policies').policyDefinitions[3].name)]",
+                        "policyDefinitionReferenceId": "Deny-PostgreSql-Server-Flexible-Server-LogAnalytics",
+                        "groupNames": [
+                            "Encryption"
+                        ],
+                        "parameters": {
+                            "effect": {
+                                "value": "[[parameters('postgreSqlServerAndFlexibleLogAnalytics')]"
                             }
```
Comment (marvinbuss) [inline foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json:517]: Some parameters are missing here.

File: foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json [hunk 7]
```diff
@@ -126,54 +422,110 @@
...
+                        "policyDefinitionId": "[concat(variables('scope'), '/providers/Microsoft.Authorization/policyDefinitions/', variables('policies').policyDefinitions[1].name)]",
+                        "policyDefinitionReferenceId": "Deny-PostgreSql-Server-Tls-Minimum-Version",
                         "groupNames": [
                             "Network"
                         ],
                         "parameters": {
                             "effect": {
-                                "value": "[[parameters('postgreSqlSsl')]"
+                                "value": "[[parameters('postgreSqlServerTlsMinimumVersion')]"
+                            }
```
Comment (marvinbuss) [inline foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json:493]: A parameter is missing here.

**Review by marvinbuss — CHANGES_REQUESTED** [review]
Please see comments below.

**Comment by marvinbuss** [comment]
Please also resolve merge conflicts.

File: foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json [hunk 8]
```diff
@@ -17,47 +17,343 @@
...
+                                "metadata": {
+                                    "displayName": "Effect",
+                                    "description": "Enable or disable the execution of the policy"
+                                },
+                                "allowedValues": [
+                                    "Deny",
+                                    "Audit",
+                                    "Disabled"
+                                    ],
+                                "defaultValue": "Deny"
```
[the remark this answers is not in the document]
Comment (sasever) [reply foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json]: fixed

File: foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json [hunk 9]
```diff
@@ -17,47 +17,343 @@
...
+                                    "equals": "Microsoft.DBforPostgreSQL/servers"
+                                },
+                                {
+                                    "field": "Microsoft.DBforPostgreSQL/servers/sslEnforcement",
+                                    "exists": "true"
+                                },
+                                {
+                                    "field": "Microsoft.DBforPostgreSQL/servers/sslEnforcement",
+                                    "notEquals": "Enabled"
+                                }
```
[the remark this answers is not in the document]
Comment (sasever) [reply foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json]: fixed with last commit

File: foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json [hunk 10]
```diff
@@ -17,47 +17,343 @@
...
+                            }
                         }
+                    },
+                    "name": "Deny-PostgreSql-Server-Minimal-TLS-Connection"
+                },
+                {
+                    "properties": {
+                        "displayName": "Enforce Customer Managed Keys for PostgreSQL Flexible database",
+                        "mode": "Indexed",
+                        "description": "Azure Database Flexible Server for PostgreSQL supports deploying a specific minimum TLS version requirement connecting your Azure Database for PostgreSQL server to client applications.",
```
[the remark this answers is not in the document]
Comment (sasever) [reply foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json]: fixed with last commit

File: foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json [hunk 11]
```diff
@@ -17,47 +17,343 @@
...
-                        }
+                        "policyRule": {
+                            "if": {
+                                "anyOf": [
+                                    {
+                                        "field": "type",
+                                        "equals": "Microsoft.DBforPostgreSQL/flexibleServers"
+                                    },
+                                    {   "field": "type", "equals": "Microsoft.DBforPostgreSQL/servers" 
+                                    }
```
[the remark this answers is not in the document]
Comment (sasever) [reply foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json]: fixed with last commit

File: foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json (hunk 4 above)
Comment (sasever) [reply foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json]: fixed with last commit

File: foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json (hunk 6 above)
Comment (sasever) [reply foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json:528]: fixed with last commit

File: foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json (hunk 7 above)
Comment (sasever) [reply foundations/azure/referenceImplementations/core/managementGroupTemplates/policyDefinitions/Compliant-PostgreSQLPolicySetDefinition.json]: fixed with last commit

**Pull request closed without merging**